### PR TITLE
feat: Add pypi actions

### PR DESCRIPTION
**Pull request by fsbraun**
## Description

Adds actions for pypi publication.

## Related resources

* #...
* #...

## Checklist

* [ ] I have opened this pull request against ``master``
* [ ] I have added or modified the tests when changing logic
* [ ] I have followed [the conventional commits guidelines](https://www.conventionalcommits.org/) to add meaningful information into the changelog
* [ ] I have read the [contribution guidelines ](https://github.com/django-cms/django-cms/blob/develop/CONTRIBUTING.rst) and I have joined #workgroup-pr-review on 
[Slack](https://www.django-cms.org/slack) to find a “pr review buddy” who is going to review my pull request.

### CODE REVIEW

**Comment by marksweb** [comment]
To fix the ruff error, the pyupgrade part of the config needs removing

**Comment by marksweb** [comment]
@fsbraun because this is going to be new to pypi, I'd need a generic token so this has got me thinking about this new trusted publishing method of deployment.

Looking at the action repo, it's actually a simpler workflow to work this way so I'd like to swap to this style.

https://github.com/pypa/gh-action-pypi-publish#trusted-publishing

It looks like the setup is this simple for live;

```
name: Publish Python 🐍 distributions 📦 to pypi

on:
  release:
    types:
      - published

jobs:
  build-n-publish:
    name: Build and publish Python 🐍 distributions 📦 to pypi
    runs-on: ubuntu-latest
    environment:
      name: pypi
      url: https://pypi.org/p/djangocms-versioning
    permissions:
      id-token: write
    steps:
    - uses: actions/checkout@v3
    - name: Set up Python 3.10
      uses: actions/setup-python@v4
      with:
        python-version: '3.10'

    - name: Install pypa/build
      run: >-
        python -m
        pip install
        build
        --user
    - name: Build a binary wheel and a source tarball
      run: >-
        python -m
        build
        --sdist
        --wheel
        --outdir dist/
        .

    - name: Publish distribution 📦 to PyPI
      if: startsWith(github.ref, 'refs/tags')
      uses: pypa/gh-action-pypi-publish@release/v1
```

And the test release...

```
name: Publish Python 🐍 distributions 📦 to TestPyPI

on:
  push:
    branches:
      - master

jobs:
  build-n-publish:
    name: Build and publish Python 🐍 distributions 📦 to TestPyPI
    runs-on: ubuntu-latest
    environment:
      name: pypi
      url: https://pypi.org/p/djangocms-versioning
    permissions:
      id-token: write
    steps:
    - uses: actions/checkout@v3
    - name: Set up Python 3.10
      uses: actions/setup-python@v4
      with:
        python-version: '3.10'

    - name: Install pypa/build
      run: >-
        python -m
        pip install
        build
        --user
    - name: Build a binary wheel and a source tarball
      run: >-
        python -m
        build
        --sdist
        --wheel
        --outdir dist/
        .

    - name: Publish distribution 📦 to Test PyPI
      uses: pypa/gh-action-pypi-publish@release/v1
      with:
        repository-url: https://test.pypi.org/legacy/
        skip_existing: true
```

**Comment by fsbraun** [comment]
@marksweb Done!

File: .github/workflows/publish-to-test-pypi.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    branches:
+      - master
+
+jobs:
+  build-n-publish:
+    name: Build and publish Python 🐍 distributions 📦 to TestPyPI
+    runs-on: ubuntu-latest
+    environment:
+      name: pypi
+      url: https://pypi.org/p/djangocms-versioning
```
Comment (marksweb) [inline .github/workflows/publish-to-test-pypi.yml:14]: This should probably be

```suggestion
      url: https:// test.pypi.org/p/djangocms-versioning
```

**Review by marksweb — CHANGES_REQUESTED** [review]

**Comment by fsbraun** [comment]
@marksweb Can you mark your comment as resolved so that I can merge it?

**Review by marksweb — APPROVED** [review]

**Pull request merged**